nengo_viz/static/viz_netgraph_item.js

```diff
@@ -0,0 +1,452 @@
+                var h = ng.get_scaled_height();    
+                var item = ng.svg_objects[uid];
+                var parent = item.parent;
+                while (parent != null) {
+                    w = w * parent.size[0] * 2;
+                    h = h * parent.size[1] * 2;
+                    parent = parent.parent;
+                }
+                item.set_position(item.pos[0] + event.dx / w, 
+                                  item.pos[1] + event.dy / h);
+            },
+            onend: function(event) {
+                var item = ng.svg_objects[uid];
+                item.constrain_position();                
+                ng.notify({act:"pos", uid:uid, x:item.pos[0], y:item.pos[1]});
+            }});
+            
+    /** dragging the edge of item to change its size */
+    interact(this.shape)
+        .resizable({
+            edges: { left: true, right: true, bottom: true, top: true }
+            })
+        .on('resizemove', function(event) {            
+            var item = ng.svg_objects[uid];
+            var w = ng.get_scaled_width();
+            var h = ng.get_scaled_height();    
+            var parent = item.parent;
+            while (parent != null) {
+                w = w * parent.size[0] * 2;
+                h = h * parent.size[1] * 2;
+                parent = parent.parent;
+            }
+            
+            item.set_size(item.size[0] + event.deltaRect.width / w / 2, 
+                          item.size[1] + event.deltaRect.height / h / 2);
+            item.set_position(item.pos[0] + event.deltaRect.width / w / 2 + 
+                                            event.deltaRect.left / w, 
+                              item.pos[1] + event.deltaRect.height / h / 2 + 
+                                            event.deltaRect.top / h);
+            })
+        .on('resizeend', function(event) {
+            var item = ng.svg_objects[uid];
+            item.constrain_position();                
+            ng.notify({act:"pos_size", uid:uid, 
+                       x:item.pos[0], y:item.pos[1],
+                       width:item.size[0], height:item.size[1]});
+            });
+            
+    interact(this.g)
+        .on('wheel', function(event) {
+            var item = ng.svg_objects[uid];
+            var step_size = 1.1; // size of zoom per wheel click
+
+            var delta = event.wheelDeltaY || -event.deltaY
+            var scale = delta > 0 ? step_size : 1.0 / step_size;
+            
+            item.set_size(item.size[0] * scale, item.size[1] * scale);
+            
+            item.constrain_position();                
+            ng.notify({act:"pos_size", uid:uid, 
+                       x:item.pos[0], y:item.pos[1],
+                       width:item.size[0], height:item.size[1]});
+                       
+            event.stopPropagation();           
+            });
+            
+    if (info.type == 'net') {
```

Comment:
When you can, use '===' instead of '==', because it's always better to have strict-er conditions